fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,195 @@
+                raise ValueError("Failed to load the gRPC protocol buffer files:\nERROR: {0}".format(error_msg))
+
+            # Converting resilient function input data into json object
+            try:
+                grpc_function_json_data = json.loads(grpc_function_data)
+            except Exception:
+                raise FunctionError(u"Function input 'grpc_function_data' must be a JSON formatted String: {0}".format(grpc_function_json_data))
+
+            # Checking Communication type
+            if _grpc_communication_type.lower().strip() == "unary":
+                try:
+                    # Connecting to gRPC Server with given Authentication type
+                    stub_class_obj, grpc_channel_obj = grpc_helper_obj.gRPC_Connect_Server(_grpc_secure_connection,
+                                                                                           grpc_channel,
+                                                                                           grpc_stub_tuple[1],
+                                                                                           _grpc_certificate_path)
+                    stub_method = grpc_helper_obj.get_method_from_stub_object(stub_class_obj, _grpc_communication_type,
+                                                                              stub_name=_grpc_rpc_stub_method_name)
+                    if grpc_function_json_data:
+                        if stub_method[1] is not None:
+                            if six.PY2:
+                                # convert unicode strings to utf-8 for python 2.7
+                                grpc_function_json_data_tmp = ast.literal_eval(json.dumps(grpc_function_json_data))
+                            else:
+                                grpc_function_json_data_tmp = grpc_function_json_data
+
+                            response_received_tmp = stub_method[1](grpc_request_tuple[1](**grpc_function_json_data_tmp))
+
+                            # Closing the gRPC Server Connection
+                            grpc_channel_obj.close()
+                        else:
+                            raise FunctionError("Stub method not found, please specify the valid stub method.")
+                    else:
+                        raise FunctionError("Please Specify the valid data input form the Resilient.")
+                except grpc.RpcError as rpc_err:
+                    status_code = rpc_err.code()
+                    log.debug("Failed to connect to gRPC Server" \
+                        "{}\n connection status :{} \n name : {}\n value : {}".format(rpc_err, rpc_err.details(),
+                                                                                      status_code.name,
+                                                                                      status_code.value))
+                    raise FunctionError("Server Connection status : {}".format(rpc_err.details()))
+                else:
```

Comment:
No need for this `else` block. Just place the log.info under line 148: `response_received_tmp = stub_method[1](grpc_request_tuple[1](**grpc_function_json_data_tmp))`

Reply:
ok